docs-source/content/userguide/cicd/choose-an-approach.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "Choose an approach"
+date: 2019-04-11T13:36:57-04:00
+draft: false
+weight: 3
+description: "How to choose an approach."
+---
+
+Let's review what we have discussed and talk about when we might want to use
+various approaches.  We can start by asking ourselves questions like these:
+
+
+- *Can you make the desired change with a configuration override?*  
+  The WebLogic Kubernetes Operator allows you to inject a number of [configuration
+  overrides]({{< relref "/userguide/managing-domains/configoverrides/_index.md" >}})
+  into your pods at startup time.  This allows you to use the same image for multiple
+  different configurations.  A good example would be changing the settings for a data
+  source, for example. You may wish to have a larger connection pool in your production
+  environment than you do in your development/test environments.  You probably also
+  want to have different credentials.  You may want to change the service name, and
+  so on.  All of these kinds of updates can be made with configuration overrides.
+  These are placed in a Kubernetes config map, that is, they are outside of the image, so
+  they do not require rebuilding the Docker image.  If all of your changes fit into
+  this category, it is probably much better to just use configuration overrides
+  instead of building a new image.  
+- *Are you only changing the WebLogic configuration, for example, deploying or updating an
+  application, changing a resource configuration in a way that is not supported by
+  configuration overrides, and such?*  
+  If your changes fit into this category, and you have used the "domain-in-image"
+  approach and the Docker layering model, then you only need to update the top layer
+  of your image.  This is relatively easy compared to making changes in lower layers.
+  You could create a new layer with the changes, or your could rebuild/replace the
+  existing top layer with a new one.  Which approach you choose depends mainly on
+  whether you need to maintain the same domain encryption keys or not.
+- *Do you need to be able to do a rolling restart?*
+  If you need to do a rolling restart, for example to maintain the availability of
+  your applications, then you need to make sure the new domain layer has the same
+  domain encryption keys.  You cannot perform a rolling restart of a domain if the
+  new members have a different encryption key.
```

Comment:
I'm not sure that it's just rolling restart that's an issue.  We don't know where other layers may use encryption keys in, say, MDS.  I think I'd rather say something more conservative about encryption keys.